### PR TITLE
=str close client upstream to signal that no more data is going to be sent

### PR DESCRIPTION
**Pull request by 2m**

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -350,12 +350,14 @@ class TcpSpec extends AkkaSpec("akka.io.tcp.windows-connection-abort-workaround-
...
-      val result = Source.maybe[ByteString]
+      val (promise, result) = Source.maybe[ByteString]
         .via(Tcp().outgoingConnection(serverAddress.getHostName, serverAddress.getPort))
-        .runFold(ByteString.empty)(_ ++ _)
+        .toMat(Sink.fold(ByteString.empty)(_ ++ _))(Keep.both)
+        .run()
 
       Await.result(result, 3.seconds) should ===(ByteString("Early response"))
 
+      promise.success(None) // close client upstream, no more data
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: oh wow, this makes sense. Why did this pass before?

**Comment by drewhk** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala (hunk 1 above)
Comment (2m) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: With the new interpreter the error was that `Source.maybe` actor what not shutdown after the test. Maybe old interpreter was more eager when signaling cancellation.

**Comment by rkuhn** [comment]
LGTM

**Pull request closed without merging**